src/mofaflex/_core/mofaflex.py

```diff
@@ -1257,6 +1359,10 @@ def _save(
             "factors": self._factors._asdict(),
             "covariates": self._covariates,
             "covariates_names": self._covariates_names,
+            "n_guiding_vars": self._n_guiding_vars,
+            "guiding_vars_names": self._guiding_vars_names,
+            "guiding_vars_factors": self._guiding_vars_factors,
+            "guiding_vars_n_categories": self._guiding_vars_n_categories,
```

Comment:
That is only used during training, it should not be an attribute of `self` and should definitely not be saved to disk.
```suggestion
```